### DISCUSSION TITLE
Max input current on PiJuiceZero - using with larger 12v Solar Panels

### DISCUSSION THREAD

**Discussion by robcoward** [PiSupply/PiJuice#1098] [General]
I have a PiJuice Zero attached to a Raspberry Pi Zero 2 W driving a USB webcam, with a 10,000mAh battery attached. My aim is to power this by Solar / battery alone, so that it can run off-grid making either a Wifi connection (or maybe later a 4G USB modem) back to my network.
Initially I tried powering it with a 6W panel with a 5v USB output, but due to the non-ideal location of the solar panel not getting direct full sunlight on it, I never had the pijuice_cli status show the USB Micro input as anything other than `NOT PRESENT` or `BAD`.

Next I tried upgrading the panel to a 30w 12v panel, theoretically capable of generating 2.5A in optimum conditions,  connected to a 12v to 5v buck converter before attaching it to the PiJuiceZero. This now occasionally now gives me a `WEAK` input and every now and then changes to `PRESENT` - however when switching off the mains input on the PIZero power connector, this immediated drops back to `WEAK` and the battery level starts to drop as the current from the panel is clearly still not enough to power the pi and keep the battery charged.

Now I want to try a larger panel but not being an electrical engineer, I need some advice from the community please. I read in the PiJuice hardware docs that the USB Micro Input has configurable current limiting, with settings at either 1.5A or 2.5A. If I use a 120W 12v solar panel, that is theoretically in optimum conditions (not where this panel is going to be used) capable of generating a 10A current. As long as the buck convertor I use to step down to 5v can support 10A, can I connect this then into the PiJuice and have it limit the current to the required 2.5A (if the panel ever generates something higher)?  

The buck convertor I am currently using has a micro usb connector output which I use to connect to the PiJuice board - would I be better off soldering the buck convertor output to the J4 pads instead of using the micro usb ? It's a shame the input only supports 4.2 - 10V - it would be a lot handier if it could take the 12v feed direct without needing to step down.

**Comment by ArjenR49** [comment]
Have you read this and follow-ups:
https://blog.rfox.eu/en/Hardware/SolarPi_experiment_1_The_PiJuice_Fiasco.html
